synapseclient/core/credentials/credential_provider.py

```diff
@@ -252,8 +269,16 @@ def get_credentials(
             [SynapseCredentials][synapseclient.core.credentials.cred_data.SynapseCredentials]
                 returned by the first non-None provider in its list, None otherwise
         """
+        selected_profile = user_login_args.profile or os.getenv("SYNAPSE_PROFILE")
         for provider in self.cred_providers:
-            creds = provider.get_synapse_credentials(syn, user_login_args)
+            creds = provider.get_synapse_credentials(
+                syn,
+                UserLoginArgs(
+                    username=user_login_args.username,
+                    auth_token=user_login_args.auth_token,
+                    profile=selected_profile,
+                ),
```

Comment:
`user_login_args` is already of type `UserLoginArgs` right? So can we just pass `user_login_args` straight in?